src/withTrackingComponentDecorator.js

```diff
@@ -118,20 +123,32 @@ export default function withTrackingComponentDecorator(
         [getTrackingDispatcher, getTrackingDataFn, getProcessFn]
       );
 
+      const propsToBePassed = useMemo(
+        () => (forwardRef ? { ...props, ref: rtFwdRef } : { ...props }),
```

Comment:
I think we can avoid spreading the props for the false branch:

```suggestion
        () => (forwardRef ? { ...props, ref: rtFwdRef } : props),
```